infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutDesignDao.kt

```diff
@@ -1,78 +1,109 @@
 package fi.fta.geoviite.infra.tracklayout
 
+import fi.fta.geoviite.infra.common.DomainId
 import fi.fta.geoviite.infra.common.IntId
+import fi.fta.geoviite.infra.logging.AccessType
+import fi.fta.geoviite.infra.logging.daoAccess
 import fi.fta.geoviite.infra.util.DaoBase
+import fi.fta.geoviite.infra.util.DbTable
 import fi.fta.geoviite.infra.util.getEnum
 import fi.fta.geoviite.infra.util.getFreeText
-import fi.fta.geoviite.infra.util.getInstant
 import fi.fta.geoviite.infra.util.getIntId
 import fi.fta.geoviite.infra.util.getLocalDate
+import fi.fta.geoviite.infra.util.getRowVersion
+import fi.fta.geoviite.infra.util.queryOne
 import fi.fta.geoviite.infra.util.setUser
 import fi.fta.geoviite.infra.util.toDbId
 import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
 import org.springframework.stereotype.Component
 import org.springframework.transaction.annotation.Transactional
-import java.sql.Timestamp
+import java.time.Instant
 
 @Component
 @Transactional(readOnly = true)
 class LayoutDesignDao(
     jdbcTemplateParam: NamedParameterJdbcTemplate?,
 ) : DaoBase(jdbcTemplateParam) {
 
+    fun fetch(id: IntId<LayoutDesign>): LayoutDesign {
+        val sql = """
+            select id, name, estimated_completion, design_state
+            from layout.design
+            where id = :id
+        """.trimIndent()
+        return jdbcTemplate.queryOne(sql, mapOf("id" to id.intValue)) { rs, _ ->
```

Comment:
Heräsin tästä taas miettimään että mikähän olisi se paras pattern näihin kun haluaa hakea tasan 1 tai 0-1 asiaa. Tämä queryOne/queryOptional tuntui aluksi hyvältä ajatukselta mutta idea ei osaa highlightata tuota sql-koodia ellei stringiin erikseen kommentoi (`//language=SQL`) sitä kieltä. 

Alempana käytetty queryForObject on sekin tavallaan oikeaa asiaa, mutta palauttaa Javan takia nullablen mutta ei koskaan oikeasti nullia, joten sillä joutuu assertoimaan non-null paluuarvon kotlinia varten ja kuitenkaan  ei saa edes `getOptional`-henkistä nullable palautusta toteutettua :roll_eyes: . 

Perus query palauttaa listan kuten olettaisi ja siitä voi poimia assertoidun ainoan arvon järkevällä virheellä `getOne(id, list)` mutta ei sekään kovin nättiä ole. 

Opinions?

Reply:
Tuo //language=SQL tuntuu kyllä ainakin hivenen kankealta 🤔, listasta toki saisi sen yhden arvon .singlellä() tai .singleOrNullilla() myös

Reply:
Tämä on kyllä ärsyttävä. Kävin katsomassa vähän mallia että miten tätä ongelmaa on taklattu muualla, ja esim. `SplitDao`:ssa ongelma oli ratkaistu siten, että on erilliset `get()`- ja `getOrThrow()`-metodit, joista ensimmäinen toimii `getOptional()`:illa, ja jälkimmäinen kutsuu ensimmäistä ja heittää käsin `NoSuchEntityException`:in mikäli vastauksena tuli null. Muuten tämä olisi patterni jonka kopsaisin tännekin, mutta en tiedä onko meillä täällä kauheasti noille non-throwing-versioille muuten kuin silloin jos Designeja käydään oikeasti poistamassa kannasta sellaisen poistamisen tai maaliin saattamisen myötä (tästähän puhuttiin eilen, mutta en kyllä nyt suoraan enää muista mihin sen osalta päädyttiin)